source/fe/fe_bernardi_raugel.cc

```diff
@@ -55,7 +55,25 @@ FE_BernardiRaugel<dim>::FE_BernardiRaugel(const unsigned int p)
 
   // const unsigned int n_dofs = this->dofs_per_cell;
 
-  this->mapping_type = mapping_none;
+  if(dim==2)
+    this->mapping_type = {mapping_none, mapping_none, 
+                          mapping_none, mapping_none,
+                          mapping_none, mapping_none, 
+                          mapping_none, mapping_none,
+                          mapping_piola, mapping_piola,
+                          mapping_piola, mapping_piola};
+  else if(dim==3)
+    this->mapping_type = {mapping_none, mapping_none, mapping_none,
+                          mapping_none, mapping_none, mapping_none,
+                          mapping_none, mapping_none, mapping_none,
+                          mapping_none, mapping_none, mapping_none,
+                          mapping_none, mapping_none, mapping_none,
+                          mapping_none, mapping_none, mapping_none,
+                          mapping_none, mapping_none, mapping_none,
+                          mapping_none, mapping_none, mapping_none,
+                          mapping_piola, mapping_piola,
+                          mapping_piola, mapping_piola,
+                          mapping_piola, mapping_piola};
```

Comment:
add the following:
```
  else
    Assert (false, ExcNotImplemented());
```
to make sure that if the code is ever used for anything other than 2d and 3d, you will be notified that that case has not been implemented.

Reply:
On line 53 I have `Assert(dim == 2 || dim == 3, ExcImpossibleInDim(dim));`

Should I still add this one as well?

Reply:
Yes, I think that would be useful. The assertion at the top is an expression of the general idea that this function isn't implemented for certain cases, whereas adding the assertion here would point to a specific place where could would need to be augmented if one wanted to extend the function.

Reply:
Okay got it!